src/composables/accountSelector.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  computed,
+  ref,
+  toRef,
+  watch,
+} from 'vue';
+
+import type {
+  IAccountSelectorEntry,
+  ITransaction,
+} from '@/types';
+import {
+  ACCOUNT_SELECT_TYPE_FILTER,
+  AccountSelectTypeFilter,
+  PROTOCOLS,
+  TX_DIRECTION,
+} from '@/constants';
+import { createCustomScopedComposable, getDefaultAccountLabel } from '@/utils';
+import {
+  useAccounts,
+  useAddressBook,
+  useLatestTransactionList,
+  useTransactionData,
+} from '@/composables';
+
+import { useAeMiddleware } from '@/protocols/aeternity/composables';
+import { useAeNames } from '@/protocols/aeternity/composables/aeNames';
+import { getInnerTransaction } from '@/protocols/aeternity/helpers';
+
+export const useAccountSelector = createCustomScopedComposable(() => {
+  const { getName } = useAeNames();
+  const { getMiddleware } = useAeMiddleware();
+
+  const latestTransactions = ref<IAccountSelectorEntry[]>([]);
+  const searchQuery = ref<string>('');
+  const {
+    addressBookFiltered: rawAddressBookFiltered,
+    addressBookFilteredByProtocol,
+    protocolFilter,
+    showBookmarked,
+    getAddressBookEntryByAddress,
+    setProtocolFilter,
+    setShowBookmarked,
+    clearFilters: addressBookClearFilters,
+  } = useAddressBook();
+  const {
+    activeAccount,
+    accountsGroupedByProtocol,
+    getAccountByAddress,
+  } = useAccounts();
+  const { accountsTransactionsLatest } = useLatestTransactionList();
+  const accountSelectType = ref<AccountSelectTypeFilter>(ACCOUNT_SELECT_TYPE_FILTER.addressBook);
+  const addressBookFiltered = computed(() => rawAddressBookFiltered.value.map((account) => ({
+    ...account,
+    isAddressBookEntry: true,
+  })));
+  const ownAddresses = computed(() => {
+    if (protocolFilter.value) {
+      return accountsGroupedByProtocol.value[protocolFilter.value]?.map((account) => (
+        {
+          name: getName(account.address).value || getDefaultAccountLabel(account),
+          address: account.address,
+          isBookmarked: false,
+          protocol: protocolFilter.value ?? PROTOCOLS.aeternity,
+          type: account.type,
+        }
+      ));
+    }
+    return [];
+  });
+  const accountsFilteredByType = computed(
+    () => {
+      switch (accountSelectType.value) {
+        case ACCOUNT_SELECT_TYPE_FILTER.bookmarked:
+        case ACCOUNT_SELECT_TYPE_FILTER.addressBook:
+          return addressBookFiltered.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.recent:
+          return latestTransactions.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.owned:
+          return ownAddresses.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.all:
+          return [...(addressBookFiltered.value ?? []), ...(ownAddresses.value ?? [])];
```

Comment:
```suggestion
          return [...addressBookFiltered.value, ...ownAddresses.value];
```
As far as I see those value couldn't be false so providing an empty arrays as a fallback makes no sense here.